src/ai/defaultai.cc

```diff
@@ -6134,6 +6132,17 @@ void DefaultAI::consider_productionsite_influence(BuildableField& field,
 		++field.collecting_producers_nearby.at(bo.get_collected_map_resource());
 	}
 
+	if (bo.type == BuildingObserver::Type::kProductionsite) {
+		const ProductionSiteDescr* productionsite = dynamic_cast<const ProductionSiteDescr*>(bo.desc);
+		for (const auto& supported : productionsite->supported_productionsites()) {
+			if (field.supporters_nearby_new.count(supported) != 1) {
```

Comment:
what is point of this != 1?

Reply:
If count() == 1, the object is a member of the container.

Please do not waste your time reviewing this pull request. It is not ready.